src/main/scala/lantern/ad_lms_vector.scala

```diff
@@ -256,199 +254,198 @@ trait TensorExp extends Dsl with Diff {
   class BackendDefault extends BackendNative
   val backend: Backend = new BackendDefault
 
-  class Tensor(val data: Rep[Array[Float]], val dimsSeq: NSeq[Int]) extends Serializable {
+  class Tensor(val data: Rep[Array[Float]], val dimensions: NSeq[Int]) extends Serializable {
```

Comment:
Should this just be called `shape`?

Reply:
[There's already something called `shape`](https://github.com/feiwang3311/Lantern/pull/13/files#diff-731c21d511864089e5fa75e2e69d72aaR266), so I named this guy `dimensions`.

The current `shape` (i.e. `TTT(dimensions)`) is meant to be used everywhere. It enables debugging, as @GSAir explains above.